// hackoregon/components/campaign/Contributors.jsx
var React = require('react/addons'),
    _ = require('lodash'),
    CurrencyFormat = require('hackoregon/utils/CurrencyFormat'),
    d3 = require('d3');

var Contributors = React.createClass({
    propTypes: {
        title: React.PropTypes.string,
        contributors: React.PropTypes.array
    },
    getDefaultProps: function() {
        return {
            title: '',
            contributors: []
        }
    },
    getBarWidth: function(amount) {
        var donorMax = d3.max(this.props.contributors, function(donor){return donor.sum;});
        var donorSize = d3.scale.linear().domain([0, donorMax]).range([0,1]);

        return donorSize(amount);
    },
    getBarStyle: function(amount) {
        return {
            width: (100 * this.getBarWidth(amount)) + '%'
        }
    },
    getContributions: function() {
        return _.map(this.props.contributors, function(contributor) {
            return (
                <div className="donor row">
                    <div className="wider col-sm-7 payee">
                        {contributor.contributor_payee}
                    </div>
                    <div className="wider col-sm-2">
                        {CurrencyFormat(contributor.sum)}
                    </div>
                    <div className="col-sm-3 hidden-xs"> 
                        <div className="contributor-donation" style={this.getBarStyle(contributor.sum)} />
                    </div>      
                </div>
            );

        }, this)
    },
    render: function() {
        return (
            <div className="contributors">
                <div className="btn btn-default col-xs-12">
                    {this.props.title}
                    <span className="badge pull-right">{this.props.contributors.length} Total</span>
                </div>
                <div className="col-sm-12 donorList">
                    {this.getContributions()}
                </div>
            </div>
        )
    }
});

module.exports = Contributors;
