// hackoregon/api/API.js
var request = require('superagent');

var BASE_REQUEST = 'http://54.213.83.132/hackoregon/http/',

    SEARCH_BASE = 'candidate_search/',
    COMMITTEE_BASE = 'committee_data_by_id/',
    TRANSACTION_BASE = 'current_candidate_transactions/',
    STATE_BASE = 'candidate_in_by_state_by_id/',
    DATE_BASE = 'candidate_sum_by_date/';

var OREGON_SUM_BASE = 'all_oregon_sum/_/',
    OREGON_CONTRIBUTIONS_BASE = 'oregon_by_contributions/_/',
    OREGON_PURPOSE_BASE = 'oregon_by_purpose_codes/_/',
    OREGON_IN_BY_BASE = 'oregon_in_by_state/_/',
    OREGON_STATE_SUM_BY_DATE_BASE = 'state_sum_by_date/_/',
    OREGON_INDIVIDUAL_CONTRIBUTORS_BASE = 'oregon_individual_contributors/_/',
    OREGON_BUSINESS_CONTRIBUTORS_BASE = 'oregon_business_contributors/_/',
    OREGON_COMMITTEE_CONTRIBUTORS_BASE = 'oregon_committee_contributors/_/';


function parseRespone(cb) {
    return function(err, res) {
        cb(err, JSON.parse(res.text));
    }
}

var API = {
    search: function(keyword, cb) {
        request.get(BASE_REQUEST + SEARCH_BASE + keyword + '/').end(parseRespone(cb));
    },
    getCommitteeDataById: function(id, cb) {
        request.get(BASE_REQUEST + COMMITTEE_BASE + id + '/').end(parseRespone(cb));
    },
    getTransactionDataById: function(id, cb) {
        request.get(BASE_REQUEST + TRANSACTION_BASE + id + '/').end(parseRespone(cb));
    },
    getStateDataById: function(id, cb) {
        request.get(BASE_REQUEST + STATE_BASE + id + '/').end(parseRespone(cb));
    },
    getDateDataById: function(id, cb) {
        request.get(BASE_REQUEST + DATE_BASE + id + '/').end(parseRespone(cb));
    },
    getAllOregonSum: function(cb) {
        request.get(BASE_REQUEST + OREGON_SUM_BASE).end(parseRespone(cb));
    },
    getOregonByContributions: function(cb) {
        request.get(BASE_REQUEST + OREGON_CONTRIBUTIONS_BASE).end(parseRespone(cb));
    },
    getOregonByPurposeCodes: function(cb) {
        request.get(BASE_REQUEST + OREGON_PURPOSE_BASE).end(parseRespone(cb));
    },
    getOregonInByState: function(cb) {
        request.get(BASE_REQUEST + OREGON_IN_BY_BASE).end(parseRespone(cb));
    },
    getStateSumByDate: function(cb) {
        request.get(BASE_REQUEST + OREGON_STATE_SUM_BY_DATE_BASE).end(parseRespone(cb));
    },
    getOregonIndividualContributors: function(cb) {
        request.get(BASE_REQUEST + OREGON_INDIVIDUAL_CONTRIBUTORS_BASE).end(parseRespone(cb));
    },
    getOregonBusinessContributors: function(cb) {
        request.get(BASE_REQUEST + OREGON_BUSINESS_CONTRIBUTORS_BASE).end(parseRespone(cb));
    },
    getOregonCommitteeContributors: function(cb) {
        request.get(BASE_REQUEST + OREGON_COMMITTEE_CONTRIBUTORS_BASE).end(parseRespone(cb));
    }
}

module.exports = API;
