// hackoregon/api/API.js
var request = require('superagent');

var BASE_REQUEST = 'http://54.213.83.132/hackoregon/http/',

    SEARCH_BASE = 'candidate_search/',
    COMMITTEE_BASE = 'committee_data_by_id/',
    TRANSACTION_BASE = 'current_candidate_transactions/',
    STATE_BASE = 'candidate_in_by_state_by_id/',
    DATE_BASE = 'candidate_sum_by_date/';

function parseRespone(cb) {
    return function(err, res) {
        cb(err, JSON.parse(res.text));
    }
}

var API = {
    search: function(keyword, cb) {
        request.get(BASE_REQUEST + SEARCH_BASE + keyword + '/').end(parseRespone(cb));
    },
    getCommitteeDataById: function(id, cb) {
        request.get(BASE_REQUEST + COMMITTEE_BASE + id + '/').end(parseRespone(cb));
    },
    getTransactionDataById: function(id, cb) {
        request.get(BASE_REQUEST + TRANSACTION_BASE + id + '/').end(parseRespone(cb));
    },
    getStateDataById: function(id, cb) {
        request.get(BASE_REQUEST + STATE_BASE + id + '/').end(parseRespone(cb));
    },
    getDateDataById: function(id, cb) {
        request.get(BASE_REQUEST + DATE_BASE + id + '/').end(parseRespone(cb));
    }
}

module.exports = API;
