// hackoregon/components/Search.jsx
var React = require('react/addons');

var SearchBox = React.createClass({
    propTypes: {
        onSubmit: React.PropTypes.func,
        value: React.PropTypes.string
    },
    getInitialState: function() {
        return {
            value: this.props.value
        }
    },
    componentWillReceiveProps: function(nextProps) {
        this.setState({
            value: nextProps.value
        });
    },
    getDefaultProps: function() {
        return {
            onSubmit: function() {},
            value: ''
        };
    },
    handleChange: function(e) { 
        this.setState({
            value: e.target.value
        });
    },
    handleKeyUp: function(e) {
        if (e.key == 'Enter') {
            this.props.onSubmit(this.state.value);
        }
    },
    handleClick: function(e) {
        this.props.onSubmit(this.state.value);
    },
    render: function() {
        return (
            <div className="container results">
              <div className="row">  
                <div className="row search-box">
                    <div className="hidden-xs col-sm-1 icon search icon-{{ searchType }}"></div>
                    <div className="col-xs-9 col-xs-offset-1 col-sm-6 col-sm-offset-0">
                      <input 
                        onKeyUp={this.handleKeyUp}
                        value={this.state.value} 
                        onChange={this.handleChange} 
                        type="text" 
                        className="form-control" 
                        placeholder="Candidate, Measure, or PAC Name" />
                      <img className="bg-search-icon" src="/images/icons/search.svg" />
                    </div>
                    <div className="col-xs-1">
                      <div className="btn icon-arrow" onClick={this.handleClick}></div>
                    </div>
                </div>
              </div>
            </div>
        )
    }
});

module.exports = SearchBox;
